gradle-baseline-java/src/test/groovy/com/palantir/baseline/JunitReportsPluginSpec.groovy

```diff
@@ -0,0 +1,59 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline
+
+import nebula.test.IntegrationSpec
+import nebula.test.functional.ExecutionResult
+import spock.lang.Unroll
+
+@Unroll
+class JunitReportsPluginSpec extends IntegrationSpec {
+    private static final List<String> GRADLE_TEST_VERSIONS = ['5.6.4', '6.1']
+
+    def '#gradleVersionNumber: configures the checkstlye plugin correctly'() {
+        setup:
+        gradleVersion = gradleVersionNumber
+
+        when:
+        buildFile << '''
+            apply plugin: 'java'
+            apply plugin: 'checkstyle'
+            apply plugin: 'com.palantir.junit-reports'
+            
+            repositories {
+                jcenter()
+            }
+        '''.stripIndent()
+
+        file('src/main/java/foo/Foo.java') << '''
+            package foo;
+            public class Foo {}
+        '''.stripIndent()
+
+        ExecutionResult executionResult = runTasks('checkstyleMain')
+
+        then:
+        // Running checkstyle inside nebula does not work as there are classpath problems that result in the `groovy-all`
+        // not being on Gradle's classpath. So the best we can do to verify that the checkstyle actually ran is to
+        // verify we get the classpath error that happens when the checkstyle class runs :(
+        // https://github.com/gradle/gradle/issues/3995
+        executionResult.standardError.contains 'java.lang.ClassNotFoundException: groovy.util.AntBuilder'
```

Comment:
would it make sense to just not use nebula for this one test? i.e. to go through all the ProjectBuilder stuff manually?

Reply:
I don't think we can run with a specified gradle version without using nebula?

Reply:
This also happens in gradle test kit and the other 3rd `gradleTest` framework :(

Reply:
you can https://docs.gradle.org/current/userguide/test_kit.html#sub:gradle-runner-gradle-version

Reply:
also if you disable debug it will not cause issues. Gradle test runner debug mode is a little to magical on your classpath

Reply:
Well, I gave [using testkit/ProjectBuilder directly a go](https://github.com/palantir/gradle-baseline/pull/1176/commits/642a72badaf0b23af1a9617628ad070ac4d58810) and I just couldn't get it to work (kept saying the existing code worked on Gradle 6.1) :( So this (or no test) may have to suffice.